flood_api/__main__.py

```diff
@@ -1,11 +1,29 @@
+from contextlib import asynccontextmanager
+
 from fastapi import FastAPI
-from flood_api.settings import Settings
-from flood_api.routers import healthcheck
 
-settings = Settings()
-app = FastAPI(title="Flood API", version=settings.version)
-app.include_router(healthcheck.router)
+from flood_api.dependencies.flooddata import fetch_flood_data, flood_data_fetcher
+from flood_api.routers import flood, healthcheck
+from flood_api.settings import settings
+
+import asyncio
+
 
+@asynccontextmanager
+async def lifespan(flood_app: FastAPI):
+    await fetch_flood_data(flood_app)
+    asyncio.create_task(flood_data_fetcher(flood_app))
+    yield
```

Comment:
Should there be something after this yield in order to shut down the task of fetching new data?

Reply:
The flood_data_fetcher-task seems to be shutting down without any problems when the main-thread is shutting down. I can't think of anything else that needs to be cleaned up